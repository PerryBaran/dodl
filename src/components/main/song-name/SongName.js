import React, { useContext, useEffect, useState } from 'react';
import style from './songName.module.css';
import AppContext from '../../AppContext';

const SongName = (props) => {
    const { songs, songIndex } = props
    const { hideWhilePlaying } = useContext(AppContext);
    const [songJustChanged, setSongJustChanged] = useState(false);

    //makes song name visible on song change
    useEffect(() => {
        setSongJustChanged(true);
        const timer = setTimeout(() => {
            setSongJustChanged(false);
        }, 1000);
        return () => clearTimeout(timer)
    }, [songIndex]);

    const addClassOnSongChange = () => {
        return songJustChanged? style.nameGlow : ''
    }

    return (
        <div className={`centerFlex positionBottom`}>
            <h2 className={`${style.name} ${hideWhilePlaying(style.pause)} ${addClassOnSongChange()}`}>{songs[songIndex].title}</h2>
        </div>
    );
}

export default SongName;