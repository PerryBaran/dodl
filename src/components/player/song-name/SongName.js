import React, { useContext, useEffect, useState } from 'react';
import style from './songName.module.css';
import AppContext from '../../../utils/context/AppContext';

const SongName = (props) => {
    const { songs, songIndex } = props
    const { isPlaying } = useContext(AppContext);
    const [songChangeClassName, setSongChangeClassName] = useState(false);

    //makes song name visible on song change
    useEffect(() => {
        setSongChangeClassName(true);
        const timer = setTimeout(() => {
            setSongChangeClassName(false);
        }, 1000);
        return () => clearTimeout(timer)
    }, [songIndex]);

    return (
        <div className={`centerFlex positionBottom`}>
            <h2 className={`${style.name} ${!isPlaying && style.pause} ${songChangeClassName && style.nameGlow}`}>{songs[songIndex].title}</h2>
        </div>
    );
}

export default SongName;