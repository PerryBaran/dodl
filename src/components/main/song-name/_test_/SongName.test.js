import React from "react";
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom'
import SongName from "../SongName";
import AppContext from "../../../AppContext";
import renderer from "react-test-renderer";

afterEach(cleanup);

const songs = [{title: 'dummy', src: true}, {title: 'text', src: true}]

const RenderWithContext = (index = 0, isPlaying = false) => {
    const hideWhilePlaying = (input) => {
        if (!isPlaying) {
          return input
        }
        return ''
    };
    return (
        <AppContext.Provider value={{hideWhilePlaying}}>
            <SongName songs={songs} songIndex={index}/>
        </AppContext.Provider>
    )
};

it("renders without crashing", () => { 
    render(RenderWithContext());
});

it("shows className while paused", () => {
    render(RenderWithContext());
    const h2 = screen.getByRole('heading');
    expect(h2).toHaveClass('pause')
});

it("hides className while playing", () => {
    render(RenderWithContext(0, true));
    const h2 = screen.getByRole('heading');
    expect(h2).not.toHaveClass('pause')
});

it("renders correct name", () => {
    render(RenderWithContext(1));
    const h2 = screen.getByRole('heading');
    expect(h2).toHaveTextContent(songs[1].title)
});

it("matches snapshot while paused", () => {
    const tree = renderer.create(RenderWithContext()).toJSON();
    expect(tree).toMatchSnapshot();
});

it("matches snapshot while playing", () => {
    const tree = renderer.create(RenderWithContext(0, true)).toJSON();
    expect(tree).toMatchSnapshot();
});

it("matches snapshot with different songIndex", () => {
    const tree = renderer.create(RenderWithContext(1)).toJSON();
    expect(tree).toMatchSnapshot();
})