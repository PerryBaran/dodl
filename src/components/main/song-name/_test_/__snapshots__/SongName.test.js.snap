// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot while paused 1`] = `
<div
  className="centerFlex positionBottom"
>
  <h2
    className="name pause "
  >
    dummy
  </h2>
</div>
`;

exports[`matches snapshot while playing 1`] = `
<div
  className="centerFlex positionBottom"
>
  <h2
    className="name  nameGlow"
  >
    dummy
  </h2>
</div>
`;

exports[`matches snapshot with different songIndex 1`] = `
<div
  className="centerFlex positionBottom"
>
  <h2
    className="name pause "
  >
    text
  </h2>
</div>
`;
